walletseed/seed.go

```diff
@@ -26,69 +26,89 @@ func GenerateRandomSeed(size uint) ([]byte, error) {
 	return hdkeychain.GenerateSeed(uint8(size))
 }
 
-// checksumByte returns the checksum byte used at the end of the seed mnemonic
-// encoding.  The "checksum" is the first byte of the double SHA256.
-func checksumByte(data []byte) byte {
-	intermediateHash := sha256.Sum256(data)
-	return sha256.Sum256(intermediateHash[:])[0]
-}
+// ------------------------------------------------------------------------
 
-// EncodeMnemonicSlice encodes a seed as a mnemonic word list.
-func EncodeMnemonicSlice(seed []byte) []string {
-	words := make([]string, len(seed)+1) // Extra word for checksumByte
-	for i, b := range seed {
-		words[i] = pgpwordlist.ByteToMnemonic(b, i)
+func bytesToBits(bytes []byte) []byte {
+	length := len(bytes)
+	bits := make([]byte, length*8)
+	for i := 0; i < length; i++ {
+		b := bytes[i]
+		for j := 0; j < 8; j++ {
+			mask := byte(1 << uint8(j))
+			bit := b & mask
+			if bit == 0 {
+				bits[(i*8)+8-(j+1)] = '0'
+			} else {
+				bits[(i*8)+8-(j+1)] = '1'
+			}
+		}
 	}
-	checksum := checksumByte(seed)
-	words[len(words)-1] = pgpwordlist.ByteToMnemonic(checksum, len(seed))
-	return words
+	return bits
+}
+
+// CheckSummed returns a bit slice of entropy with an appended check sum
+func CheckSummed(ent []byte) []byte {
+	cs := CheckSum(ent)
+	bits := bytesToBits(ent)
+	return append(bits, cs...)
+}
+
+// CheckSum returns a slice of bits from the given entropy
+func CheckSum(ent []byte) []byte {
+	h := sha256.New()
+	h.Write(ent) // nolint: errcheck
+	cs := h.Sum(nil)
```

Comment:
Is this for copying?

Reply:
"Always pass nil to hash.Sum(b []byte)"
http://golang-examples.tumblr.com/post/86164939039/always-pass-nil-to-hashsumb-byte